website/versioned_docs/version-v1.x/circuits.md

```diff
@@ -7,45 +7,407 @@ sidebar_position: 7
-   state and ballot trees according to the contents of said messages.
-2. `TallyVotes.circom`, which counts votes from users' ballots, batch by batch.
-3. `Subsidy.circom`, which implements [pairwise subsidy](https://hackmd.io/@chaosma/H1_9xmT2K)
+1. [`ProcessMessages.circom`](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/processMessages.circom), which takes a batch of encrypted messages, decrypts them, and generates a proof that the coordinator's local processing was performed correctly.
+2. [`TallyVotes.circom`](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/tallyVotes.circom), which counts votes from users' ballots, batch by batch.
+3. [`Subsidy.circom`](https://github.com/privacy-scaling-explorations/maci/blob/dev/circuits/circom/subsidy.circom), which implements [pairwise subsidy](https://hackmd.io/@chaosma/H1_9xmT2K).
 
-Each circuit is parameterised and it is important to set the right parameters
-to your use case. For example, if you want to support up to 3125 messages, the message tree depth parameter should be set to `5` (as $5^5 = 3125$).
+Each circuit is parameterised and it is important to set the right parameters that matches your use case. For example, if you want to support up to 3125 messages, the message tree depth parameter should be set to `5` (as $5^5 = 3125$).
 
-Next, navigate to the `cli/` directory and edit `zkeys.config.yml`.
+## Background
 
-This config file defines the parameters required for MACI's circuits.
+### zk-SNARKs
 
-### Message processing
+MACI uses zk-SNARKs to prove that the coordinator has correctly processed a batch of messages and that the votes have been tallied correctly. zk-SNARKs are a type of zero-knowledge proof which allows a prover to prove that they know a secret without revealing the secret itself. In MACI, the prover is the coordinator. Users, can at any point after the proof generation, verify that the process was done correctly.
+
+### Circom
+
+MACI's circuits are written using [Circom](https://docs.circom.io/), a domain-specific language (DSL) used to write zk-SNARK circuits. Circom syntax resembles JavaScript, and it currently is one of the most popular DSL in use by zk developers. Please refer to their documentation to learn more about the language.
+
+### Proving system
+
+MACI uses [GROTH16](https://eprint.iacr.org/2016/260.pdf) as its proving system. GROTH16 is a zk-SNARK proving system that allows for the generation of proofs that are small and fast to verify.
+
+## How are the circuits used
+
+The circuits are used by the coordinator (the prover) to prove that they have correctly processed a batch of messages and tallied the votes correctly. This happens after a Poll has completed, and the coordinator has merged the state and message trees. The coordinator then generates a proof for each batch of messages, and submits them to the contract. The contract then verifies the proofs and updates the commitments on chain.
```

Comment:
> correctly processed a batch of messages and tallied the votes correctly

General terminology question - can/should "messages" & "votes" be used interchangeably? Or are votes a specific type of message (i.e. messages can be key switch commands or vote commands)?

Reply:
Correct, votes are a specific type of messages. A message could simply be a topup message requesting to topup a participant's voice credits